plugins/net_plugin/net_plugin.cpp

```diff
@@ -1143,6 +1146,11 @@ namespace eosio {
    }
 
 
+   template<typename Function>
+   void connections_manager::for_each_supplied_peer( Function&& f ) const {
+      std::for_each(supplied_peers.begin(), supplied_peers.end(), std::forward<Function>(f));
```

Comment:
A `shared_lock` on `connections_mtx` is required here.

Reply:
Added.